src/EditorFeatures/TestUtilities2/Intellisense/ModernCompletionTestState.vb

```diff
@@ -183,24 +182,33 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
 
         Public Overrides Function CompletionItemsContainsAny(displayText As String, displayTextSuffix As String) As Boolean
             AssertNoAsynchronousOperationsRunning()
-            Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(TextView)
-            Assert.NotNull(session)
-            Dim items = session.GetComputedItems(CancellationToken.None)
+            Dim items = GetComputedItemsOpt(TextView)
+            Assert.NotNull(items)
 
             Return items.Items.Any(Function(i) i.DisplayText = displayText AndAlso i.Suffix = displayTextSuffix)
         End Function
 
         Public Overrides Sub AssertItemsInOrder(expectedOrder As String())
             AssertNoAsynchronousOperationsRunning()
-            Dim session = GetExportedValue(Of IAsyncCompletionBroker)().GetSession(TextView)
-            Assert.NotNull(session)
-            Dim items = session.GetComputedItems(CancellationToken.None).Items
+            Dim compoutedItems = GetComputedItemsOpt(TextView)
+            Assert.NotNull(compoutedItems)
+            Dim items = compoutedItems.Items
             Assert.Equal(expectedOrder.Count, items.Count)
             For i = 0 To expectedOrder.Count - 1
                 Assert.Equal(expectedOrder(i), items(i).DisplayText)
             Next
         End Sub
 
+        Public Overrides Async Function WaitForAsynchronousOperationsAsync(view As ITextView) As Task
+            Await Task.WhenAll(MyBase.WaitForAsynchronousOperationsAsync(), Task.Run(Sub()
+                                                                                         Dim items = GetComputedItemsOpt(TextView)
+                                                                                     End Sub))
```

Comment:
What's going on here? Why is this stronger than just waiting for asynchronous operations directly?

Reply:
This went away in the latest version, where we don't need to separate legacy and modern completion.